### PR TITLE
Set connection counterparty while processing connOpenAck

### PR DESCRIPTION
**Pull request by hu55a1n1**
Closes: #1532 

Thanks for all the help, @ancazamfir!

<del>Note: there's a `TODO` in the code that we must figure out how to resolve before this is merged.</del>
______

For contributor use:

- [x] Added a changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [x] If applicable: Unit tests written, added test to CI.
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Updated relevant documentation (`docs/`) and code comments.
- [x] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

File: modules/src/core/ics03_connection/handler/conn_open_ack.rs [hunk 1]
```diff
@@ -33,6 +33,7 @@ pub(crate) fn process(
     let counterparty_matches =
         if let Some(counterparty_connection_id) = conn_end.counterparty().connection_id() {
             &msg.counterparty_connection_id == counterparty_connection_id
+            // TODO(hu55a1n1): check client_id as well
```
[the remark this answers is not in the document]
Comment (hu55a1n1) [reply modules/src/core/ics03_connection/handler/conn_open_ack.rs]: Done! 👍 https://github.com/informalsystems/ibc-rs/pull/1533/commits/a7ca446cc91c1549f4a43f3d7b4c3d50409d5e34

File: .changelog/unreleased/bug-fixes/ibc/1532-connOpenAck-counterparty-conn-id-not-set.md [hunk 2]
```diff
@@ -0,0 +1 @@
+- Set the `counterparty_connection_id` correctly in the ICS03 `connOpenAck` handler (i.e. `ics03_connection::handler::conn_open_ack::process()`).
```
Comment (romac) [inline .changelog/unreleased/bug-fixes/ibc/1532-connOpenAck-counterparty-conn-id-not-set.md:1]:
```suggestion
- Set the `counterparty_connection_id` correctly in the ICS03 [`connOpenAck` handler](https://github.com/informalsystems/ibc-rs/blob/master/modules/src/core/ics03_connection/handler/conn_open_ack.rs)
   ([#1532](https://github.com/informalsystems/ibc-rs/issues/1532))
```

File: modules/tests/support/model_based/ICS03.tla [hunk 3]
```diff
@@ -331,7 +331,8 @@ ICS03_ConnectionOpenAck(
                         \* verification passed; update the connection state to
                         \* "Open"
                         LET updatedConnection == [connection EXCEPT
-                            !.state = "Open"
+                            !.state = "Open",
+                            !.counterpartyConnectionId = counterpartyConnectionId
```
Comment (hu55a1n1) [inline modules/tests/support/model_based/ICS03.tla:335]: I tested this change with the old-code (as per @romac's suggestion) and the MBT tests do catch the missing `set_counterparty()` and fail.

Comment (romac) [reply modules/tests/support/model_based/ICS03.tla:335]: Awesome :)

**Review by romac — APPROVED** [review]
🌸

**Pull request merged**